indexing-service/src/main/java/io/druid/indexing/common/task/CompactionTask.java

```diff
@@ -292,9 +302,14 @@ private static DataSchema createDataSchema(
     final List<QueryableIndex> queryableIndices = loadSegments(timelineSegments, segmentFileMap, indexIO);
 
     // find merged aggregators
+    for (QueryableIndex index : queryableIndices) {
+      if (index.getMetadata() == null) {
+        throw new RE("Index metadata doesn't exist for interval[%s]", index.getDataInterval());
```

Comment:
The error message should really include the segment identifier. Imagine the poor user that gets this error and can't figure out which segment to look at.

Perhaps we can keep it along with the QueryableIndex using a Pair or a small static holder class returned from `loadSegments`.

Reply:
Good point. Added segment identifier.